elasticsearch/client.go

```diff
@@ -18,81 +18,88 @@
 package elasticsearch
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"errors"
 	"io"
+	"io/ioutil"
 	"net/http"
+	"net/url"
+	"strings"
 
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/version"
 
-	v7 "github.com/elastic/go-elasticsearch/v7"
-	v7esapi "github.com/elastic/go-elasticsearch/v7/esapi"
+	esv7 "github.com/elastic/go-elasticsearch/v7"
 
-	v8 "github.com/elastic/go-elasticsearch/v8"
-	v8esapi "github.com/elastic/go-elasticsearch/v8/esapi"
+	esv8 "github.com/elastic/go-elasticsearch/v8"
 )
 
 // Client is an interface designed to abstract away version differences between elasticsearch clients
 type Client interface {
+	// TODO: deprecate
 	// Search performs a query against the given index with the given body
 	Search(index string, body io.Reader) (int, io.ReadCloser, error)
-	SecurityHasPrivilegesRequest(body io.Reader, header http.Header) (int, io.ReadCloser, error)
+	// Makes a request with application/json Content-Type and Accept headers by default
+	// pass/overwrite headers with "key: value" format
+	JSONRequest(method, path string, body interface{}, headers ...string) JSONResponse
 }
 
 type clientV8 struct {
-	client *v8.Client
+	v8 *esv8.Client
```

Comment:
I guess I disagree. For me `client.v8` is more informative and readable than `client.client`

Reply:
It would be `clientV8.client` or in the receiver methods `c.client` vs with the change `c.v8`, where `v8` could be anything, and one needs to check the struct definition to verify what it is. 

This discussion has been going back and forth already for some time, keep the changes if you feel that strongly.

Reply:
cool, ill close this one then